controller/controller.go

```diff
@@ -258,6 +261,80 @@ func ScaleUp(logger logrus.FieldLogger, mr *models.MixedMetricsReporter, db pgin
 	return creationErr
 }
 
+func ScaleDown(logger logrus.FieldLogger, mr *models.MixedMetricsReporter, db pginterfaces.DB, redisClient redisinterfaces.RedisClient, clientset kubernetes.Interface, scheduler *models.Scheduler, amount, timeoutSec int) error {
+	l := logger.WithFields(logrus.Fields{
+		"source":    "scaleDown",
+		"scheduler": scheduler.Name,
+		"amount":    amount,
+	})
+	timeout := make(chan bool, 1)
+	go func() {
+		time.Sleep(time.Duration(timeoutSec) * time.Second)
+		timeout <- true
+	}()
+
+	l.Debug("accessing redis")
+	pipe := redisClient.TxPipeline()
+	sReady := pipe.SPopN(models.GetRoomStatusSetRedisKey(scheduler.Name, models.StatusReady), int64(amount))
+	_, err := pipe.Exec()
+	if err != nil {
+		l.WithError(err).Error("scale down error")
```

Comment:
use a better message here so we can distinguish this error from the one below: "failed to retrieve ready rooms from redis" maybe?